src/NSwag.CodeGeneration.CSharp/Templates/Client.Class.liquid

```diff
@@ -58,6 +58,53 @@
     partial void PrepareRequest({{ HttpClientType }} client, System.Net.Http.HttpRequestMessage request, System.Text.StringBuilder urlBuilder);
     partial void ProcessResponse({{ HttpClientType }} client, System.Net.Http.HttpResponseMessage response);
 
+    protected struct ObjectResponseResult<T>
+    {
+        public ObjectResponseResult(T responseObject, string responseText)
+        {
+            this.ResponseObject = responseObject;
+            this.ResponseText = responseText;
+        }
+
+        public T      ResponseObject { get; }
+        public string ResponseText   { get; }
+    }
+
+    protected virtual async System.Threading.Tasks.Task<ObjectResponseResult<T>> ReadObjectResponseAsync<T>(System.Net.Http.HttpResponseMessage response, System.Collections.Generic.IReadOnlyDictionary<string,System.Collections.Generic.IEnumerable<string>> responseHeadersAsDictionary)
+    {
+#if DEBUG
```

Comment:
Do we really need DEBUG code here?

Reply:
That's intentional - to allow developers to inspect the server responses when debugging if there's an incorrect server response.

I've changed it to a different symbol (`NSWAG_STRING_RESPONSES`) to be clearer, so users have to explicitly opt-in to this less efficient code-path by enabling the symbol in their project's source code.